File: data-automation-project/backend/data-automation-project/src/main/java/com/studies/lvb/data_automation_project/dto/UserDTO.java
package com.studies.lvb.data_automation_project.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO implements Serializable {

    private String name;

    private String email;

    private String password;
}

File: ws-user-login/ws-user-login-api/src/main/kotlin/com/lvb/studys/kotlin/api/wsuserlogin/dto/UserDTO.kt
package com.lvb.studys.kotlin.api.wsuserlogin.dto


import com.lvb.studys.kotlin.api.wsuserlogin.entity.Role
import com.lvb.studys.kotlin.api.wsuserlogin.entity.User
import java.io.Serializable

data class UserDTO(

    var id: String? = null,

    var firstName: String? = null,

    var lastName: String? = null,

    var email: String? = null,

    var roles: List<Role> = ArrayList()

) : Serializable {

    //TODO remove this and create a generic converter for entity to DTO/ DTO to entity
    constructor(user: User) : this() {
        this.id = user.id
        this.firstName = user.firstName
        this.lastName = user.lastName
        this.email = user.email
        this.roles = user.roles
    }
}

